src/DependencyInjection/DoctrineMigrationsExtension.php

```diff
@@ -90,15 +90,21 @@ public function load(array $configs, ContainerBuilder $container): void
             $diDefinition->addMethodCall('setDefinition', [$doctrineId, new Reference($symfonyId)]);
         }
 
-        if (! isset($config['services'][MetadataStorage::class])) {
+        if (isset($config['services'][MetadataStorage::class])) {
+            $container->removeDefinition('doctrine_migrations.schema_filter_listener');
```

Comment:
When a user defines their own storage class, there is no telling what they are going to do… they might even not store it in a database. At this point they're one their own.